lib/geordi/chromedriver_updater.rb

```diff
@@ -62,36 +65,65 @@ def update_chromedriver(latest_chromedriver_version)
-      response = Net::HTTP.get_response(uri)
-
-      if response.is_a?(Net::HTTPSuccess)
-        file = Tempfile.new(['chromedriver', '.zip'])
+    def download_chromedriver(version)
+      fetch_response(chromedriver_url(version), "Could not download chromedriver v#{version}.") do |response|
+        file = Tempfile.new(%w[chromedriver .zip])
         file.write(response.body)
 
         file
-      else
-        Interaction.fail("Could not download chromedriver v#{latest_version}.")
       end
     end
 
-    def latest_version(chrome_version)
-      return @latest_version if @latest_version
-
-      uri = URI("https://chromedriver.storage.googleapis.com/LATEST_RELEASE_#{major_version(chrome_version)}")
+    def fetch_response(url, error_message)
+      uri = URI(url)
       response = Net::HTTP.get_response(uri)
 
       if response.is_a?(Net::HTTPSuccess)
-        @latest_version = response.body.to_s
+        yield(response)
       else
-        Interaction.fail("Could not download the chromedriver v#{chrome_version}.")
+        Interaction.fail(error_message)
+      end
+    end
+
+    def chromedriver_url(chrome_version)
+        chromedriver_per_platform = chromedriver_download_data.dig("milestones", milestone_version(chrome_version), "downloads", "chromedriver")
+        chromedriver = chromedriver_per_platform&.find do |chromedriver|
+          chromedriver["platform"] == "linux64"
+        end
+
+        if chromedriver && chromedriver["url"]
+          chromedriver["url"]
+        else
+          Interaction.fail("Could not find chromedriver download url for chrome version v#{chrome_version}")
+        end
+    end
+
+    def chromedriver_download_data
+      return @chromedriver_download_data if @chromedriver_download_data
+
+      fetch_response(VERSIONS_PER_MILESTONES_URL, "Could not find chromedriver download data") do |response|
+        begin
+          chromedriver_download_data = JSON.parse(response.body)
+        rescue JSON::ParserError
+          Interaction.fail("Could not parse chromedriver download data")
+        end
+        @chromedriver_download_data = chromedriver_download_data
       end
     end
 
+    def latest_version(chrome_version)
+      latest_version = chromedriver_download_data.dig("milestones", milestone_version(chrome_version), "version")
+      latest_version || Interaction.fail("Could not parse chromedriver download data")
+    end
+
     def unzip(zip, output_dir)
       _stdout_str, _error_str, status = Open3.capture3('unzip', '-d', output_dir, '-o', zip.path)
 
+      # the archive contains a folder in which the relevant files are located. These files must be copied to ~/bin.
+      FileUtils.mv("#{output_dir}/chromedriver-linux64/chromedriver", output_dir)
+      FileUtils.mv("#{output_dir}/chromedriver-linux64//LICENSE.chromedriver", output_dir)
+      FileUtils.rm_rf("#{output_dir}/chromedriver-linux64")
```

Comment:
This should happen after the success check below. Also, the comment should say "must be moved", not copied.